great_expectations/analytics/base_event.py

```diff
@@ -32,25 +33,25 @@ class Event:
     action: Action
 
     @property
-    def data_context_id(self) -> UUID:
+    def data_context_id(self) -> UUID | None:
         return get_config().data_context_id
 
     @property
     def organization_id(self) -> UUID | None:
         return get_config().organization_id
 
     @property
-    def oss_id(self) -> UUID:
+    def oss_id(self) -> UUID | None:
         return get_config().oss_id
 
     @property
     def user_id(self) -> UUID | None:
         return get_config().user_id
 
     @property
-    def distinct_id(self) -> UUID:
+    def distinct_id(self) -> UUID | None:
```

Comment:
Since `oss_id` and `user_id` can now return `None` this type also needs to be updated. This means, we may send `""` to post-hog. Is that ok? I also tried raising if this was going to be `None` but we have at least 1 test where it is: `tests/analytics/test_analytics.py::test_event_identifiers[analytics_config1]`